WolvenKit.App/Services/UpdateService.cs

```diff
@@ -0,0 +1,168 @@
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Net.Http;
+using System.Security.Cryptography;
+using System.Threading.Tasks;
+using Semver;
+using WolvenKit.App.Helpers;
+using WolvenKit.App.Models;
+using WolvenKit.Common;
+using WolvenKit.Core.Interfaces;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace WolvenKit.App.Services;
+
+public class UpdateService : IUpdateService
+{
+    private readonly ILoggerService _loggerService;
+    private readonly ISettingsManager _settingsManager;
+    private readonly HttpClient _httpClient;
+    
+    public UpdateService(ILoggerService loggerService, ISettingsManager settingsManager)
+    {
+        _loggerService = loggerService;
+        _settingsManager = settingsManager;
+        _httpClient = new HttpClient();
+        _httpClient.DefaultRequestHeaders.Add("User-Agent", "WolvenKit");
+        _httpClient.DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json");
+    }
+    
+    public async Task UpdateToNewestVersion()
+    {
+        var latestRelease = await GetLatestRelease();
+        if (latestRelease is null)
+        {
+            return;
+        }
+        
+        if (!await IsUpdateAvailable(latestRelease))
+        {
+            return;
+        }
+        
+        var tempPath = Path.Join(Path.GetTempPath(), "WolvenKitUpdate");
+        Directory.CreateDirectory(tempPath);
+        var downloadZipPath = Path.Join(tempPath, latestRelease.TagName + ".zip");
+        var portableAsset = latestRelease.Assets.First(a => a.Name == $"WolvenKit-{latestRelease.TagName}.zip");
+        
+        try
+        {
+            var responseAsset = await _httpClient.GetAsync(portableAsset.DownloadUrl);
+            responseAsset.EnsureSuccessStatusCode();
+            await File.WriteAllBytesAsync(downloadZipPath, await responseAsset.Content.ReadAsByteArrayAsync());
+        }
+        catch (HttpRequestException ex)
+        {
+            _loggerService.Error($"Failed to download update from: {portableAsset.DownloadUrl}");
+            _loggerService.Error(ex);
+            return;
+        }
+        
+        if (portableAsset.Digest?.Split(":")[^1] != BitConverter.ToString(SHA256.Create().ComputeHash(File.ReadAllBytes(downloadZipPath))).Replace("-", "").ToLowerInvariant())
+        {
+            Directory.Delete(tempPath, true);
+            _loggerService.Error("Downloaded update asset is invalid! Aborting update.");
+            return;       
+        }
+        
+        var unzipPath = Path.Join(tempPath, "unzip");
+        Directory.CreateDirectory(unzipPath);
+        ZipFile.ExtractToDirectory(downloadZipPath, unzipPath);
+        
+        var exePath = Environment.ProcessPath ?? Path.Combine(AppContext.BaseDirectory, "WolvenKit.exe");
+        var scriptPath = Path.Combine(tempPath, "update.ps1");
+        var vbsScriptPath = Path.Combine(tempPath, "update.vbs");
+        File.WriteAllText(scriptPath, $@"
```

Comment:
You should be really careful with this, AV software might flag the whole WKit if you run files like this.

Or, they could even block the execution of these scripts until they are analyzed in their clouds.